File: src/main/java/me/leodev/trollingfreedomreborn/commands/Spin.java
package me.leodev.trollingfreedomreborn.commands;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;

import java.util.ArrayList;

public class Spin implements Listener {

    public static ArrayList<String> spin1 = new ArrayList<>();


    public static void Spin(final Player p) {
        spin1.add(p.getName());
        final Location tp = p.getLocation();
        int id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, new Runnable() {
            public void run() {
                tp.setYaw(tp.getYaw() + 90.0F);
                p.teleport(tp);
            }
        }, 10L, 5L);
        Core.instance.addTask(p,"spin",id);
    }

    public void StopSpin(Player p) {
        spin1.remove(p.getName());
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/classics/Spin.kt
package com.leomadrassi.trollingfreedomreborn.trolls.classics

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.Listener

class Spin : Listener {

    companion object {
        val spin1 = mutableListOf<String>()

        fun Spin(p: Player) {
            spin1.add(p.name)
            val tp = p.location
            val id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, Runnable {
                tp.setYaw(tp.yaw + 90.0f)
                p.teleport(tp)
            }, 10L, 5L)
            Core.instance.addTask(p, "spin", id)
        }
    }

    fun StopSpin(p: Player) {
        spin1.remove(p.name)
    }
}
